--- a/ngx_rtmp_eval.c
+++ b/ngx_rtmp_eval.c
@@ -30,0 +31,15 @@
+static void
+ngx_rtmp_eval_connection_number(void *ctx, ngx_rtmp_eval_t *e,
+    ngx_str_t *ret)
+{
+    ngx_rtmp_session_t  *s = ctx;
+
+    ret->data = ngx_pnalloc(s->connection->pool, NGX_INT64_LEN);
+    if (ret->data == NULL) {
+        ret->len = 0;
+        return;
+    }
+    ret->len = ngx_sprintf(ret->data, "%ui", s->connection->number) - ret->data;
+}
+
+
@@ -55,0 +71,4 @@
+
+    { ngx_string("clientid"),
+      ngx_rtmp_eval_connection_number,
+      0 },
